clients/src/main/java/org/apache/kafka/common/utils/ByteUtils.java

```diff
@@ -253,6 +253,26 @@ public static long readVarlong(ByteBuffer buffer)  {
         return (value >>> 1) ^ -(value & 1);
     }
 
+    /**
+     * Read a double-precision 64-bit format IEEE 754 value.
+     *
+     * @param in The input to read from
+     * @return The double value read
+     */
+    public static double readDouble(DataInput in) throws IOException {
+        return Double.longBitsToDouble(in.readLong());
```

Comment:
I was checking protobuf and they do:

```java
    public double readDouble() throws IOException {
      return Double.longBitsToDouble(readRawLittleEndian64());
    }
```

Reply:
From reading ByteBuffer, I couldn't find any documentation that strictly specified the byte format, though it'd be hard to image it ever being changed or differ across JDK implementations. DataInput, on the other hand, does specify. For both, I've maintained the wrapper and test, but used their native calls read/getDouble and write/putDouble.

Let me know if you'd prefer I fold them away altogether.

Reply:
Looks fine to me.